cmd/site-controller/deploy-watch-all-ns.yaml

```diff
@@ -77,6 +83,7 @@ rules:
 - apiGroups:
   - rbac.authorization.k8s.io
   resources:
+  - clusterrolebindings
```

Comment:
What do we actually need here?

Reply:
Can we detect if the skupper policy CRD exists without having permission to read it?

Reply:
Ideally, if the CRD is installed, nothing will work if the service controller does not have permission to read policy. Otherwise you can avoid the policy entirely by initialising without the necessary permissions.

Reply:
But of course, if the CRD is not installed, then ideally everything would work as before.

Reply:
> Can we detect if the skupper policy CRD exists without having permission to read it?

No. Today policy component check if it has permission to read it and if it is defined. In case
any of the validations is false, policy will be considered as disabled.

>Ideally, if the CRD is installed, nothing will work if the service controller does not have permission to read policy. Otherwise you can avoid the policy entirely by initialising without the necessary permissions.

That is how it is supposed to work. If permission is revoked policy will be considered as disabled.

Reply:
> Today policy component check if it has permission to read it and if it is defined. In case any of the validations is false, policy will be considered as disabled.

I think we need to change that. Could we make permission to read the CRD whether or not it exists (not a sepecific CR) a strict requirement? That way we would always be able to tell if it is installed or not. If not policy is disabled, if it is policy is enforced and inability to read a matching policy whether due to permission or anything else, would prevent the operation in question.

This would mean of course that you would have to have the extra permission to use skupper even without policy. However as is, it seems policy cannot be reliably enforced.

Reply:
I think if the CRD is defined, whether or not we have permission to read any associated CRs, we should apply policy. In other words the test for whether policy is enabled should simply be the discovery of the CRD. E.g. the test would be something like this:

```
func isPolicyEnabled(cli *VanClient) bool {
	dc, err := discovery.NewDiscoveryClientForConfig(cli.RestConfig)
	resources, err := dc.ServerResourcesForGroupVersion("skupper.io/v1alpha1")
	if errors.IsNotFound(err) {
		log.Printf("Policy is NOT enabled")
		return false
	} else if err != nil {
		log.Printf("Cannot determine if policy is enabled: %s", err)
		return true
	} else {
		for _, r := range resources.APIResources {
			if r.Kind == "SkupperClusterPolicy" {
				log.Printf("Policy IS enabled")
				return true
			}
		}
		log.Printf("Policy is NOT enabled")
		return false
	}
}
```

That way, once the CRD is installed, you require the cluster role to be able to use skupper at all. If you don't have it, it is there are no matching rules that can be read and so nothing is enabled.

Reply:
@grs yes, it works just fine! Thanks for advising. I will update the main policy component to use
the discovery client instead and enforce policy in spite of permission being granted.